tests/unit/ipv6_test.cc

```diff
@@ -103,3 +103,18 @@ SEASTAR_TEST_CASE(tcp_packet_test) {
     });
 }
 
+SEASTAR_TEST_CASE(ipv6_equal_test) {
+	ipv6_addr addr1{"abcd:fedc:ba98:7654:3210:0123:4567:89ab"};
+	ipv6_addr addr2{"0123:4567:89ab:cdef:3210:0123:4567:89ab"};
+	ipv6_addr addr3{"abcd:fedc:ba98:7654:3210:0123:4567:89ab"};
+
+	socket_address sock_addr1(addr1);
+	socket_address sock_addr2(addr2);
+	socket_address sock_addr3(addr3);
+
+	BOOST_REQUIRE(sock_addr1 != sock_addr2);
+	BOOST_REQUIRE(sock_addr1 == sock_addr3);
```

Comment:
the fix in `socket_address::operator==()` is correct. but this test is not able to reproduce the issue you are fixing in this change. if we take a look at the layout of `sockaddr_in6`, see https://github.com/lattera/glibc/blob/895ef79e04a953cac1493863bcae29ad85657ee1/inet/netinet/in.h#L252-L259 . the `sin6_addr` sits in the middle of the struct. 

without your fix, both of the comparison statements return the expected result. because instead comparing the IPv6 address (16 bytes, 4 uint32_t numbers), they are comparing the following fields

- `sa_family_t` : unsigned short (uint16_t)
- `sin6_port`: uint16_t 
- `sin6_flowinfo`: uint32_t
- the lower 64 bits in `sin6_addr`

so, in order to verify your change, we have to create a test case where the lower bits of the compared `sockaddr_6` are identical, while its `sin6_addr`'s higher 32 bits are different.

so i'd suggest

1. specify the port number explicitly. could define a constant instead of a magic number for better maintainability.
2. define yet another `ipv6_addr`, like `ipv6_addr addr5{"abcd:fedc:ba98:7654:3210:0123:4567:89ae", port)`, so that its higher bits in address field is different from `addr1`.
3. use `BOOST_CHECK_NE` and `BOOST_CHECK_EQUAL` instead of `BOOST_REQUIRE`. because:

   - if the test fails, Boot.Test would be able to print out the string representation of the lhs and rhs of the compare operation
   - we don't need to fail the whole test case if a single compare fails, if the succeeding tests do not depend on the completion of previous test
4. use spaces instead tabs for indent. see https://github.com/scylladb/seastar/blob/master/coding-style.md#whitespace

Reply:
also, please add a "net:" prefix in the title of the commit message. the the output of `git log src/net/socket_address.cc` for more examples.